### ISSUE TITLE
Retrieve current progress in scenario (to be linked with front end)

### ISSUE THREAD

**Issue by nickykay98** [UoaWDCC/VPS#13]
[**User Story**]
"As a user I'd like to view my progress as I complete the simulation"

Using data for total number of scenes in the simulation and the current scene, return the percentage (or value) for how much has been completed for this to then be linked into front end implementation

**Comment by nickykay98** [comment]
@UoaWDCC/team-vps A question for all of you after messaging Matthew about this:

Currently it's very hard to know the progression of a scenario. Any given scenario isn't guaranteed to have a linear progression because scenes can be skipped, e.g., a scenario has 10 scenes total but scene 2 has a button that goes to scene 3 and a button that goes to scene 4, if the button that goes to scene 4 is pressed then incrementing the progress by 1 would be incorrect, and we also don't know to increment it by 2 because we don't know how many scenes are skipped. There are also many other scenarios that could complicate things. Matthew has had a look through MongoDB and scenes are not currently stored with a scene 'number' right now (I can confirm this is true), which would need to be implemented.

If anyone has any ideas about how to go about this or alternative ways to track progress, please comment!

**Comment by mlai962** [comment]
Possible solutions to keeping track of a scenario's progress:
- Make changes to scene entries in the database that records which scenes can come before or after that scene based on the buttons on that scene. Use this information, along with the same info for related scenes to calculate how far away a user is from the final scene. 
- Restrict the ways users can create scenarios so that they can only move forward or backward by one scene (or some other restrictions to make scenarios more straightforward) and all scenes must be used in a scenario (impractical).
- Find all possible paths a user can take in a scenario (gets complicated if a user can go to a scene that they've already been to) and adapt the progress bar (or some other form of indicating progress) depending on the path.

**Comment by nickykay98** [comment]
@Harshal-D This has been moved into Sprint 2, Matthew will continue working on this and if it doesn't take long to do, Matthew could hop onto one of the other two stories

@mlai962 Added to your list above (the last one is the one

**Comment by mlai962** [comment]
A possible implementation:
- After looking at the MongoDB further I've found out that each scenario entry in the database has an array attached to it listing out the IDs of the scenes related to that scenario. 
- If we can identify which scene a user is on and which scene the user switches to then we can use this array to determine how many scenes are left for the user to see and then to calculate the progress.
- We would need to have a temporary database entry that keeps track of the scenes already seen, and then delete this entry once the user is done with this specific play through. 
- If multiple users can play the same scenario at the same time then we need a way of differentiating the temporary database entries.

**Comment by nickykay98** [comment]
Thanks @mlai962! As discussed today, we'll put this on hold given that it's turning out to be more complex than anticipated. @Harshal-D and I will discuss your proposed implementation with the client and see what they think and how it could match with their definition of "progress" and then we'll continue this in a future sprint - for now, I've assigned you to help with https://github.com/UoaWDCC/VPS/issues/22... details/updates from today's meeting regarding that feature will be added soon!

**Comment by nickykay98** [comment]
Worked on in other sprints and branches by other people
